ts/src/bitget.ts

```diff
@@ -4069,6 +3950,12 @@ export default class bitget extends Exchange {
             // on bitget hedge mode if the position is long the side is always buy, and if the position is short the side is always sell
             // so the side of the reduceOnly order is inversed
         }
+        const orderType = this.safeString (order, 'orderType');
+        const isBuyMarket = (side === 'buy') && (orderType === 'market');
+        if (isBuyMarket) {
+            // as noted in top comment, for 'buy market' the 'size' field is COST, not AMOUNT
```

Comment:
u sure this is the case for swap?

Reply:
the size was unset for swap too and then from safeOrder set again, but right, i've added `spot` only in clause